### DISCUSSION TITLE
valor.original zerado no Payload do location

### DISCUSSION THREAD

**Discussion by LeoBOrtigoza-zz** [bacen/pix-api#363] [Q&A]
Olá, tudo bem?

Podemos usar o valor.original zerado no payload do location para que não tenha um valor sugerido (não necessariamente vamos ter um valor sugerido)?
Dessa forma, é possível dar abertura para o pagador realizar o pagamento de qualquer valor sem que tenhamos um experiencia ruim.

Ex de cenário: 
Sou um vendedor ambulante e vendo diversos tipos de produtos e tenho um único QRCode. Se tivermos um valor sugerido, podemos confundir o pagador e gerar uma experiencia ruim. Dessa forma, o melhor caminho é deixar zerado e padronizado.

**Comment by renatofrota** [comment]
QR Codes estáticos podem ser criados sem valor predeterminado. Portanto, para fundamentar a sua sugestão é preciso um detalhamento que inclua um cenário de exemplo que justifique a decisão pelo uso do QR dinâmico em vez do QR estático.

**Reply by ninrod** [reply]
Olá @LeoBOrtigoza . 

> Podemos usar o valor.original zerado no payload do location para que não tenha um valor sugerido (não necessariamente vamos ter um valor sugerido)?

Só pode vir valor zerado no payload se `modalidadeAlteracao` estiver atribuído como `1`. Caso contrário, o valor tem que ser maior do que zero.

**Reply by renatofrota** [reply]
Obrigado, ninrod. Passou totalmente batido por mim essa possibilidade. Vou me inteirar das novidades antes de opinar de novo por aqui para não atrapalhar.

**Reply by LeoBOrtigoza-zz** [reply]
Muito obrigado!! 

valeu pessoal.

**Reply by LeoBOrtigoza-zz** [reply]
Boa tarde @ninrod, tudo bem?

Estava pensando, não seria interessante ter no QR Tester as validações do campo `modalidadeAlteracao`?
2 novos cenários 
1 Sucesso - `ModalidadeAlteracao` = 1 e `Valor.original` = 0
1 Falha - `ModalidadeAlteracao` = 0 e `valor.original` = 0

Faz sentido? 

Aguardo retorno.

Obrigado.

**Reply by ninrod** [reply]
Faz sentido sim, é uma sugestão pertinente. Mas não sei se há tempo hábil para colocar mais esses testes neste momento da homologação. Talvez na próxima homologação que provavelmente acontecerá por conta do "saque pix".

**Reply by LeoBOrtigoza-zz** [reply]
Obrigado pelo retorno @ninrod! 

O canal de sugestão seria esse mesmo?

Me preocupo porque temos muitos clientes nossos em produção com esse modelo. Tenho medo de algum PSP criar uma trava para esse cenário. 
Uma outra sugestão seria criar casos de testes sugestivo e não obrigatório. Dessa forma, seria mais uma conscientização e não uma regra. 

Independente, vamos criar monitores aqui para acompanhar na produção. 

Agradeço! Ótima semana

**Reply by ninrod** [reply]
> O canal de sugestão seria esse mesmo?

Não é o canal adequado. O caminho é o pix-operacional. 

> Me preocupo porque temos muitos clientes nossos em produção com esse modelo. Tenho medo de algum PSP criar uma trava para esse cenário.
Uma outra sugestão seria criar casos de testes sugestivo e não obrigatório. Dessa forma, seria mais uma conscientização e não uma regra.

Também acho pertinente. Vou conversar com o DECEM e levar a questão. 

> Agradeço! Ótima semana

Obrigado, ótima semana para vocês também.

**Reply by LeoBOrtigoza-zz** [reply]
Show @ninrod!

Pode me atualizar aqui caso tenha notícias.

obrigado pela ajuda.

abraços.
